### PR TITLE
Updated the regex and added a test for detecting public cert for SNI

### PR DESCRIPTION
**Pull request by KarishmaGhiya**
We noticed there was a bug in @azure/msal-node in the regex used for detecting public certificate, which was breaking the SNI certificates
- Updated the regex for correctly detecting and extracting public certificate
- Added a unit test for detecting public cert for SNI
- Fixes https://github.com/Azure/azure-sdk-for-js/issues/21606

### CODE REVIEW

File: lib/msal-node/test/client/ClientAssertion.spec.ts [hunk 1]
```diff
@@ -81,5 +81,9 @@ describe('Client assertion test', () => {
         expect(ClientAssertion.parseCertificate(testCert)).toEqual([]);
     })
 
+    test('parseCertificate finds all valid certs for SNI certificate', () => {
+    const parsedCert = ['line1line2line3line4line5','line1line2line3line4line5','line1line2line3line4line5'];
+    expect(ClientAssertion.parseCertificate(TEST_CONSTANTS.SNI_CERTIFICATE)).toEqual(parsedCert);
```
Comment (jasonnutter) [inline lib/msal-node/test/client/ClientAssertion.spec.ts]:
```suggestion
        const parsedCert = ['line1line2line3line4line5','line1line2line3line4line5','line1line2line3line4line5'];
        expect(ClientAssertion.parseCertificate(TEST_CONSTANTS.SNI_CERTIFICATE)).toEqual(parsedCert);
```

**Comment by jasonnutter** [comment]
Thanks for the PR. Please run `npm run beachball:change` in the root of the repo and push the generated change file, thanks!

File: change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "type": "patch",
+  "comment": {
+    "title": "",
```
Comment (bmahall) [inline change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json]: Would be nice to have a title for more descriptive change

**Review by bmahall — APPROVED** [review]
LGTM , a change proposed in the change file

File: change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json (hunk 2 above)
Comment (sameerag) [reply change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json]: Our usual format is "<<desc>>  #PR"

**Review by sameerag — APPROVED** [review]
Lgtm, Thanks for the PR. Do we need to test for all formats of EOL (windows vs mac generated files etc)?

File: change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json (hunk 2 above)
Comment (KarishmaGhiya) [reply change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json]:
```suggestion
    "title": "Updated the regex and added a test for detecting public cert for SNI",
```

File: change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "type": "patch",
+  "comment": {
+    "title": "Updated the regex and added a test for detecting public cert for SNI",
+    "value": ""
+  },
```
Comment (jasonnutter) [inline change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json]:
```suggestion
  "comment": "Updated the regex and added a test for detecting public cert for SNI #4790",
```

**Comment by KarishmaGhiya** [comment]
@sameerag Yes we can test it on MAC generated files too. In the regex we have \r*\n so we are covering Unix cases too. we should be good

File: change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json (hunk 2 above)
Comment (KarishmaGhiya) [reply change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json]:
```suggestion
    "title": "Updated the regex and added a test for detecting public cert for SNI #4790",
```

File: change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "type": "patch",
+  "comment": {
+    "title": "",
+    "value": ""
```
Comment (KarishmaGhiya) [inline change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json]:
```suggestion

```

File: change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "type": "patch",
+  "comment": {
+    "title": "",
```
Comment (KarishmaGhiya) [inline change/@azure-msal-node-16f35a40-8506-4ab8-a9ed-cc62c026a5e2.json]:
```suggestion
    "title": "Updated the regex and added a test for detecting public cert for SNI #4790"
```

**Comment by KarishmaGhiya** [comment]
Can someone merge this PR? I don't have permissions @bmahall @sameerag 
A customer is waiting for this, wanted to make sure we release this for June.

**Review by sameerag — APPROVED** [review]

**Pull request merged**